### DISCUSSION TITLE
[Bug?] Workflow reverts to old version when opened with different bonsai version

### DISCUSSION THREAD

**Discussion by eabhorrocks** [bonsai-rx/bonsai#1736] [General]
Hi, I'm having a weird issue trying to update a workflow to run on Bonsai 2.8.2. When I load the workflow it loads a different configuration of nodes. Strangely, it seems to be loading a previous version of the workflow. I've added images of the workflow which looks fine in 2.7.2 but loads completely differently in 2.8.2. Again, this difference comes from simply loading the workflow with different  Bonsai versions. I've also added a zip containing the workflow. 

Doing a direct copy+paste between Bonsai versions causes the same behaviour.

I only noticed this as the incorrect version of the workflow has an obvious error. I'm unsure if other parts of the workflow suffer from similar issues at the moment. I tried to identify if there was some sort of config file causing this but couldn't eliminate the bug.

Any help appreciated!

![bonsaiVersionBug_272_works](https://github.com/bonsai-rx/bonsai/assets/36156626/e49986b1-6369-4966-a53a-f4de07ba1db6)
![bonsaiVersionBug_282_weirdOldVersion](https://github.com/bonsai-rx/bonsai/assets/36156626/009fbd7d-a3ea-4ad5-9a10-79a5cb414be1)
[CreateTextures_weirdCopyBug.zip](https://github.com/bonsai-rx/bonsai/files/14978782/CreateTextures_weirdCopyBug.zip)

**Comment by bruno-f-cruz** [comment]
This seems like the intended behavior. When you open a workflow in a new version of bonsai/packages, it attempts to update all nodes (including sometimes deprecating) to the newest version.

**Reply by eabhorrocks** [reply]
Thanks for the reply Bruno. 

I’m still quite confused about why a more recent version of bonsai would revert to some previous version of the workflow. I didn’t even realise previous versions of a workflow are somehow kept saved after a workflow is updated. 

In any case, would you have any advice about updating this workflow to work with 2.8.x?

thanks

**Reply by bruno-f-cruz** [reply]
Ok I will try to take another look. Can you send me the Bonsai.Config files of the two versions you are using?

**Reply by eabhorrocks** [reply]
[BonsaiConfigFiles.zip](https://github.com/bonsai-rx/bonsai/files/14981540/BonsaiConfigFiles.zip)

I've added the config files for the 2 different bonsai versions. Let me know if there's anything else.

Thanks a lot
